common/src/main/java/com/hedera/block/common/utils/FileUtils.java

```diff
@@ -0,0 +1,130 @@
+            PosixFilePermissions.asFileAttribute(
+                    Set.of(
+                            PosixFilePermission.OWNER_READ,
+                            PosixFilePermission.OWNER_WRITE,
+                            PosixFilePermission.OWNER_EXECUTE,
+                            PosixFilePermission.GROUP_READ,
+                            PosixFilePermission.GROUP_EXECUTE,
+                            PosixFilePermission.OTHERS_READ,
+                            PosixFilePermission.OTHERS_EXECUTE));
+
+    /**
+     * Use this to create a Dir or File if it does not exist with the given permissions and log the
+     * result.
+     *
+     * @param toCreate valid, non-null instance of {@link Path} to be created
+     * @param logLevel valid, non-null instance of {@link System.Logger.Level} to use
+     * @param perms valid, non-null instance of {@link FileAttribute} permissions to use when
+     *     creating the path
+     * @param semanticPathName valid, non-blank {@link String} used for logging that represents the
+     *     desired path semantically
+     * @param createDir {@link Boolean} value if we should create a directory or a file
+     * @throws IOException if the path cannot be created
+     */
+    public static void createPathIfNotExists(
+            @NonNull final Path toCreate,
+            @NonNull final System.Logger.Level logLevel,
+            @NonNull final FileAttribute<Set<PosixFilePermission>> perms,
+            @NonNull final String semanticPathName,
+            final boolean createDir)
+            throws IOException {
+        Objects.requireNonNull(toCreate);
+        Objects.requireNonNull(logLevel);
+        Objects.requireNonNull(perms);
+        StringUtils.requireNotBlank(semanticPathName);
+        final String type = createDir ? "directory" : "file";
+        if (Files.notExists(toCreate)) {
+            if (createDir) {
+                Files.createDirectories(toCreate, perms);
+            } else {
+                Files.createFile(toCreate, perms);
+            }
+            LOGGER.log(logLevel, "Created " + type + " [" + semanticPathName + "] at:" + toCreate);
+        } else {
+            LOGGER.log(
+                    logLevel,
+                    "Using existing " + type + " [" + semanticPathName + "] at:" + toCreate);
+        }
+    }
+
+    /**
+     * Read a GZIP file and return the content as a byte array.
+     *
+     * @param filePath Path to the GZIP file
+     * @return byte array of the content of the GZIP file
+     * @throws IOException if an I/O error occurs
+     */
+    public static byte[] readGzFile(@NonNull final Path filePath) throws IOException {
+        try (final InputStream fileInputStream =
+                        Files.newInputStream(Objects.requireNonNull(filePath));
+                final GZIPInputStream gzipInputStream = new GZIPInputStream(fileInputStream)) {
+            return gzipInputStream.readAllBytes();
```

Comment:
Noting to some potential dangers here, if the GZ file is too large, it could be an issue!

Reply:
You might add a comment to the storage epic @AlfredoG87 is doing. It should be addressed in a different PR

Reply:
Methods renamed with "unsafe" in method name.
Also documented both the OOME that can be thrown, and the file size limit imposed by reading into a byte array.

Reply:
Noted